src/main/java/io/strimzi/test/container/KafkaVersionService.java

```diff
@@ -61,35 +50,48 @@ public static KafkaVersionService getInstance() {
     }
 
     /**
-     * Get complete image name from given Kafka and Strimzi image version
-     * If any of the given versions is {@code null} this method fetches the latest release versions using
-     * {@link KafkaVersionService}.
+     * Get whole image and if in {@link System#getProperties()} is specified field {@code kafka.custom.image} then we use
+     * custom image. Moreover, if {@code kafkaVersion} is {@code null} this method fetches the latest release versions using
+     * {@link KafkaVersionService#getInstance()} instance.
      *
-     * @param strimziBaseImage Strimzi base image name
-     * @param strimziTestContainerImageVersion Strimzi test container image version
      * @param kafkaVersion Kafka version
-     * @return complete image name
+     *
+     * @throws UnknownKafkaVersionException when Strimzi test container does not support that specified Kafka version
+     * @return strimzi test container image path or custom image if Java property {@code kafka.custom.image} is specified
      */
-    public static String strimziTestContainerImageName(String strimziBaseImage, String strimziTestContainerImageVersion, String kafkaVersion) {
-        if (strimziBaseImage == null || strimziBaseImage.isEmpty()) {
-            strimziBaseImage = STRIMZI_BASE_IMAGE;
-        }
-        if (strimziTestContainerImageVersion == null || strimziTestContainerImageVersion.isEmpty()) {
-            strimziTestContainerImageVersion = KafkaVersionService.getInstance().latestRelease().getStrimziTestContainerVersion();
-            LOGGER.info("No Strimzi test container version specified. Using latest release:{}", strimziTestContainerImageVersion);
-        }
-
-        if (kafkaVersion == null || kafkaVersion.isEmpty()) {
-            kafkaVersion = KafkaVersionService.getInstance().latestRelease().getVersion();
-            LOGGER.info("No Kafka version specified. Using latest release:{}", kafkaVersion);
+    protected static String strimziTestContainerImageName(String kafkaVersion) {
+        final String imageName;
+        final Object strimziCustomImageName = System.getProperties().get("kafka.custom.image");
```

Comment:
You can probably define it somewhere as a constant. I also wonder little bit if the name is the right nme. Should it have some Strimzi prefix? `strimzi.test-container.custom.image` or something like that to make it more distinguishable?

Reply:
@tombentley What do you think about this? :)

Reply:
Okay I will replaced it with `strimzi.test-container.kafka.custom.image` Thanks guys :)